pkg/plugins/golang/v4/api.go

```diff
@@ -108,6 +108,9 @@ func (p *createAPISubcommand) BindFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&p.options.DoController, "controller", true,
 		"if set, generate the controller without prompting the user")
 	p.controllerFlag = fs.Lookup("controller")
+
+	fs.StringVar(&p.options.ExternalAPIPath, "external-api-path", "",
+		"Specify the Go package path for external APIs to scaffold controllers for resources not defined in this project.")
```

Comment:
> The resource flags --group, --version, --kind flags are actived by the CLI create commands whenever there is a plugin that implements the RequiresResource interface. The resource is then injected to the plugin. Doesn't make sense to me that now every plugin that wants to work with external resource needs to add explicit flags and related handling code itself.

You are speaking here about the helper:

```go
func bindResourceFlags(fs *pflag.FlagSet) *resourceOptions {
	options := &resourceOptions{}

	fs.StringVar(&options.Group, "group", "", "resource Group")
	fs.StringVar(&options.Version, "version", "", "resource Version")
	fs.StringVar(&options.Kind, "kind", "", "resource Kind")

	return options
}
```

However, not all plugins will that scaffold base on GKV can or will necessarily support External Types.
For example, we do not add this support for the Deploy Image plugin (https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go). That would change how its scaffolds works and might not make sense. 

Therefore, we cannot add the helper because not necessary any implementation of subCommands to create API/webhook will or should support external types.

Reply:
Anyway, my point here, after we add for webhooks then we can revisit it if we think that is valid.

Reply:
Yes, makes sense.